04_setup_ironic.sh

```diff
@@ -111,18 +111,18 @@ for IMAGE in ${IRONIC_IMAGE} ${VBMC_IMAGE} ${SUSHY_TOOLS_IMAGE} ; do
 done
 
 CACHED_MACHINE_OS_IMAGE="${IRONIC_DATA_DIR}/html/images/${MACHINE_OS_IMAGE_NAME}"
-if [ ! -f "${CACHED_MACHINE_OS_IMAGE}" ]; then
+if [ ! -f "${CACHED_MACHINE_OS_IMAGE}" -o ! -f "${CACHED_MACHINE_OS_IMAGE}.sha256sum" ]; then
   curl -g --insecure -L -o "${CACHED_MACHINE_OS_IMAGE}" "${MACHINE_OS_IMAGE_URL}"
```

Comment:
We could add `|| ( rm -f "${CACHED_MACHINE_OS_IMAGE}" ; exit 1 )` on the end here to ensure that the image file gets deleted if the download doesn't complete, rather than going by the non-existence of the checksum file.

Reply:
I'm not sure that will necessarily be robust, e.g if the script is killed mid-download we won't hit the error-path `rm`

We could perhaps add `-C` to curl to continue in the event of a partial download, but it's perhaps safest to just remove any existing file and download again, in the event we didn't reach the point of validating/writing the checksum file?